### PR TITLE
Allow to Modify Group in Extended view

### PR DESCRIPTION
**Pull request by ashu8912**

### CODE REVIEW

**Comment by ashu8912** [comment]
fixes #51

**Comment by ashu8912** [comment]
ezgif com-video-to-gif (1)

File: frontend/src/js/components/Groups/ItemExtended.react.js [hunk 1]
```diff
@@ -27,7 +28,7 @@ function ItemExtended(props) {
   const [application, setApplication] = React.useState(null);
   const [group, setGroup] = React.useState(null);
   const classes = useStyles();
-
+
```
Comment (joaquimrocha) [inline frontend/src/js/components/Groups/ItemExtended.react.js]: Remove this.

File: frontend/src/js/components/Groups/ItemExtended.react.js [hunk 2]
```diff
@@ -45,6 +46,9 @@ function ItemExtended(props) {
       setGroup(groupFound);
     }
   }
+  function updateGroup() {
```
Comment (joaquimrocha) [inline frontend/src/js/components/Groups/ItemExtended.react.js]: Add a newline before the function declaration.

File: frontend/src/js/components/Groups/ItemExtended.react.js [hunk 3]
```diff
@@ -70,7 +74,14 @@ function ItemExtended(props) {
                 cardMainLinkLabel={group ? group.name : '…'}
                 cardId={group ? group.id : '…'}
                 cardDescription={group ? group.description : ''}
-              />
+              >
+                <MoreMenu options={[
+                  {
+                  'label': 'Edit',
+                  'action': updateGroup,
```
Comment (joaquimrocha) [inline frontend/src/js/components/Groups/ItemExtended.react.js]: Ident these lines.

File: frontend/src/js/components/Layouts/GroupLayout.react.js [hunk 4]
```diff
@@ -3,19 +3,22 @@ import React from "react"
...
+import EditDialog from "../Groups/EditDialog";
 
 class GroupLayout extends React.Component {
 
  constructor(props) {
     super(props);
     this.onChange = this.onChange.bind(this);
-
+    this.openUpdateGroupModal=this.openUpdateGroupModal.bind(this);
+    this.closeUpdateGroupModal=this.closeUpdateGroupModal.bind(this)
```
Comment (joaquimrocha) [inline frontend/src/js/components/Layouts/GroupLayout.react.js]: Add a space around the `=` ops.

File: frontend/src/js/components/Layouts/GroupLayout.react.js [hunk 5]
```diff
@@ -36,23 +39,32 @@ class GroupLayout extends React.Component {
       applications: applicationsStore.getCachedApplications()
     })
   }
-
+  openUpdateGroupModal() {
```
Comment (joaquimrocha) [inline frontend/src/js/components/Layouts/GroupLayout.react.js]: Add a newline before this.

File: frontend/src/js/components/Layouts/GroupLayout.react.js [hunk 6]
```diff
@@ -36,23 +39,32 @@ class GroupLayout extends React.Component {
...
+    this.setState({updateGroupModalVisible: true})
+  }
+  closeUpdateGroupModal() {
```
Comment (joaquimrocha) [inline frontend/src/js/components/Layouts/GroupLayout.react.js]: Add a newline before this.

File: frontend/src/js/components/Layouts/GroupLayout.react.js [hunk 7]
```diff
@@ -36,23 +39,32 @@ class GroupLayout extends React.Component {
...
+    this.setState({updateGroupModalVisible: false})
+  }
```
Comment (joaquimrocha) [inline frontend/src/js/components/Layouts/GroupLayout.react.js]: Add a newline after this.

File: frontend/src/js/components/Layouts/GroupLayout.react.js [hunk 8]
```diff
@@ -36,23 +39,32 @@ class GroupLayout extends React.Component {
...
   render() {
     let appName = "",
         groupName = ""
 
     let applications = this.state.applications ? this.state.applications : [],
         application = _.findWhere(applications, {id: this.state.appID})
-
+    let groups=[];
+    let channels=[];
```
Comment (joaquimrocha) [inline frontend/src/js/components/Layouts/GroupLayout.react.js]: Add spaces around `=`.

File: frontend/src/js/components/Layouts/GroupLayout.react.js [hunk 9]
```diff
@@ -36,23 +39,32 @@ class GroupLayout extends React.Component {
...
     if (application) {
       appName = application.name
-
+      groups = application.groups; 
+      channels = application.channels ? application.channels : []
```
Comment (joaquimrocha) [inline frontend/src/js/components/Layouts/GroupLayout.react.js]: It's missing a `;` at the end.

File: frontend/src/js/components/Layouts/GroupLayout.react.js [hunk 10]
```diff
@@ -36,23 +39,32 @@ class GroupLayout extends React.Component {
...
       let group = _.findWhere(application.groups, {id: this.state.groupID})
       if (group) {
         groupName = group.name
       }
     }
-
+   
+    const groupToUpdate =_.findWhere(groups, {id: this.state.groupID});
+    console.log(groupToUpdate)
```
Comment (joaquimrocha) [inline frontend/src/js/components/Layouts/GroupLayout.react.js]: Remove this log.

File: frontend/src/js/components/Layouts/GroupLayout.react.js [hunk 11]
```diff
@@ -36,23 +39,32 @@ class GroupLayout extends React.Component {
...
+      groups = application.groups; 
+      channels = application.channels ? application.channels : []
       let group = _.findWhere(application.groups, {id: this.state.groupID})
       if (group) {
         groupName = group.name
       }
     }
-
+   
+    const groupToUpdate =_.findWhere(groups, {id: this.state.groupID});
```
Comment (joaquimrocha) [inline frontend/src/js/components/Layouts/GroupLayout.react.js]: Add a space after the `=`.

File: frontend/src/js/components/Layouts/GroupLayout.react.js [hunk 12]
```diff
@@ -68,7 +80,9 @@ class GroupLayout extends React.Component {
             }
           ]}
         />
-        <GroupExtended appID={this.state.appID} groupID={this.state.groupID} />
+        <GroupExtended appID={this.state.appID} groupID={this.state.groupID} handleUpdateGroup={this.openUpdateGroupModal}/>
+        <EditDialog data={{group: groupToUpdate, channels: channels}} show={this.state.updateGroupModalVisible} onHide={this.closeUpdateGroupModal} />
+
```
Comment (joaquimrocha) [inline frontend/src/js/components/Layouts/GroupLayout.react.js]: Remove this empty line.

File: frontend/src/js/components/Layouts/GroupLayout.react.js [hunk 13]
```diff
@@ -36,23 +39,34 @@ class GroupLayout extends React.Component {
...
     let applications = this.state.applications ? this.state.applications : [],
         application = _.findWhere(applications, {id: this.state.appID})
-
+    let groups = [];
+    let channels = [];
     if (application) {
       appName = application.name
-
+      groups = application.groups; 
+      channels = application.channels ? application.channels : [];
```
Comment (joaquimrocha) [inline frontend/src/js/components/Layouts/GroupLayout.react.js]: Keep the newline that was there before, i.e. between `channels` now and `let group`

File: frontend/src/js/components/Layouts/GroupLayout.react.js [hunk 14]
```diff
@@ -36,23 +39,34 @@ class GroupLayout extends React.Component {
...
       let group = _.findWhere(application.groups, {id: this.state.groupID})
       if (group) {
         groupName = group.name
       }
     }
-
+   
+    const groupToUpdate = _.findWhere(groups, {id: this.state.groupID});
```
Comment (joaquimrocha) [inline frontend/src/js/components/Layouts/GroupLayout.react.js]: Add a newline between this and the return.

File: frontend/src/js/components/Layouts/GroupLayout.react.js [hunk 15]
```diff
@@ -3,19 +3,22 @@ import React from "react"
...
+import EditDialog from "../Groups/EditDialog";
 
 class GroupLayout extends React.Component {
 
  constructor(props) {
     super(props);
     this.onChange = this.onChange.bind(this);
-
+    this.openUpdateGroupModal = this.openUpdateGroupModal.bind(this);
+    this.closeUpdateGroupModal = this.closeUpdateGroupModal.bind(this)
```
Comment (joaquimrocha) [inline frontend/src/js/components/Layouts/GroupLayout.react.js]: Keep the newline before the let below.

File: frontend/src/js/components/Groups/ItemExtended.react.js [hunk 16]
```diff
@@ -27,7 +28,6 @@ function ItemExtended(props) {
   const [application, setApplication] = React.useState(null);
   const [group, setGroup] = React.useState(null);
   const classes = useStyles();
-
```
Comment (joaquimrocha) [inline frontend/src/js/components/Groups/ItemExtended.react.js]: Don't remove the newline.

**Pull request merged**